### PR TITLE
Added functionality to read zipped cubes

### PR DESCRIPTION
**Pull request by oliver-s-lee**

### CODE REVIEW

File: cclib/parser/logfilewrapper.py [hunk 1]
```diff
@@ -250,6 +262,12 @@ def readline(self) -> str:
         """
         return next(self)
 
+    def readlines(self) -> str:
```
Comment (berquist) [inline cclib/parser/logfilewrapper.py:265]:
```suggestion
    def readlines(self) -> List[str]:
```

File: test/method/testvolume.py [hunk 2]
```diff
@@ -104,3 +104,8 @@ def test_roundtrip_cube(self):
         density_recovered = volume.read_from_cube("coarsewater.cube")
 
         assert_allclose(density.data, density_recovered.data, rtol=0.05)
+
+    def test_zip_cube(self):
+        """Check we can read from a zipped file."""
+        data = volume.read_from_cube(f"{os.path.dirname(os.path.realpath(__file__))}/co.cube.zip")
```
Comment (berquist) [inline test/method/testvolume.py:110]: We should use `pathlib.Path` in as many places as possible:
```suggestion
        data = volume.read_from_cube(Path(__file__).resolve().parent / "co.cube.zip")
```

**Review by berquist — CHANGES_REQUESTED** [review]
Minor nits

File: test/method/testvolume.py (hunk 2 above)
Comment (oliver-s-lee) [reply test/method/testvolume.py:110]: I'm a pathlib fanboy so no complaints from me

**Review by berquist — APPROVED** [review]
It's awesome that it was basically as simple as reusing `FileWrapper`.

**Pull request merged**